test/e2e/client_test.go

```diff
@@ -178,3 +188,40 @@ func (t *testClient) dumpInfo() error {
 	}
 	return nil
 }
+
+func (t *testClient) instantiateTemplate(tpl string) (*routev1.Route, error) {
```

Comment:
I dont think so but returning route from template initiation function does not look right. First, if there is more than 1 route? If there are no routes at all? If the route is named differently (L221).

Reply:
Agreed it's a weird signature. I wouldn't add this as a method in testClient but would probably keep it as a function as close to the test that is using it as possible.